src/middlewares/AuthorizationMiddleware.ts

```diff
@@ -0,0 +1,34 @@
+import UserService from "../services/UserService";
+import User, { UserRole } from "../entities/User";
+import HttpException from "../exceptions/HttpException";
+import { IRequestHandler } from "../types/IRequestHandler";
+import { IAuthorizedRequest } from "../types/IAuthorizedRequest";
+
+
+
+const AuthorizationMiddleware = (
+): IRequestHandler  => 
+  async (req: IAuthorizedRequest, res, next): Promise<User> => {
+
+    // Given 
+    let user;
+    const userService = new UserService(); 
+
+    // Validate
+    console.log(req.auth);
+    if (req.auth && req.auth.sub) {
+      user = await userService.getByAuth0_id(req.auth.sub);
```

Comment:
We should not query for a user in an authorization middleware. This is the responsibility of authentication middleware. Authorization should only check whether there's a user in the request or not and if it is then validate role permissions.